cosi/protocol/helper_functions.go

```diff
@@ -0,0 +1,138 @@
+package protocol
+
+import (
+	"fmt"
+
+	"github.com/dedis/kyber"
+	"github.com/dedis/kyber/sign/cosi"
+	"github.com/dedis/onet"
+	"github.com/dedis/onet/log"
+	"github.com/dedis/onet/network"
+)
+
+// generateCommitmentAndAggregate generates a personal secret and commitment
+// and returns respectively the secret, an aggregated commitment and an aggregated mask
+func generateCommitmentAndAggregate(s cosi.Suite, t *onet.TreeNodeInstance, publics []kyber.Point,
+	structCommitments []StructCommitment) (kyber.Scalar, kyber.Point, *cosi.Mask, error) {
```

Comment:
Go types usually don't have "struct" in them.

Reply:
This is somewhat of a convention in cothority... you have to wrap the types in a struct with onet.TreeNode to make it work with protocol channels, see https://github.com/dedis/cothority_template/blob/master/protocol/struct.go#L22

I remember Linus expressing a need to fix this but we never go around doing a major onet refactor.

Reply:
Oh sorry, I knew this, duh.